enhanced-lottie.framerfx/code/EnhancedLottie.tsx

```diff
@@ -1,134 +1,215 @@
+                        setAnimationData(data)
+                    })
+                    .catch(e => {
+                        console.error(e)
+                        console.log(
+                            "Could not parse a valid JSON from the Lottie URL"
+                        )
+                        setError(true)
+                    })
+            })
+            .catch(e => {
+                setError(true)
 
-        console.error(e)
-      })
-  }
+                console.error(e)
+            })
+    }
 
-  useEffect(() => {
-    loadLottieData()
-  }, [])
+    useEffect(() => {
+        loadLottieData()
+    }, [lottieJsonURL, loop])
 
-  useEffect(() => {
-    if (playState == 'pause') setIsPaused(true)
-    else if (playState == 'stop') setIsStopped(true)
-    else if (playState == 'play') {
-      setIsPaused(false)
-      setIsStopped(false)
-    }
-  }, [playState])
+    useEffect(() => {
+        if (playState == "pause") setIsPaused(true)
+        else if (playState == "stop") setIsStopped(true)
+        else if (playState == "play") {
+            setIsPaused(false)
+            setIsStopped(false)
+        }
+    }, [playState])
 
-  const defaultOptions = {
-    animationData: animationData,
-    loop: loop,
-    rendererSettings: {
-      preserveAspectRatio: 'xMidYMid slice',
-    },
-  }
+    let defaultOptions = {
+        animationData: animationData,
+        loop: loop,
+        rendererSettings: {
+            preserveAspectRatio: "xMidYMid slice",
+        },
+    }
 
-  return (
-    <Frame width="100%" height="100%" background="" {...props}>
-      <Lottie
-        options={defaultOptions}
-        isStopped={isStopped}
-        isPaused={isPaused}
-        speed={speed}
-        direction={direction == 'normal' ? 1 : -1}
-        currentFrame={currentFrame}
-      />
-    </Frame>
-  )
+    return (
+        <Frame width="100%" height="100%" background="" {...props}>
+            <Lottie
+                options={defaultOptions}
+                isStopped={isStopped}
+                isPaused={isPaused}
+                speed={speed}
+                direction={direction == "normal" ? 1 : -1}
+                percentage={percentage / 100}
+                segments={[segmentStart, segmentEnd]}
+                playSegments={playSegments}
+                eventListeners={[
+                    {
+                        eventName: "complete",
+                        callback: onComplete,
+                    },
+                    {
+                        eventName: "loopComplete",
+                        callback: onLoopComplete,
+                    },
+                    {
+                        eventName: "enterFrame",
+                        callback: onEnterFrame,
+                    },
+                    {
+                        eventName: "segmentStart",
+                        callback: onSegmentStart,
+                    },
+                ]}
```

Comment:
Add segment controls and Event Listeners which can be used to programmatically control the Lottie animation, e.g. through Overrides (the PropertyControl/Framer interface is somewhat unfinished, as it relies on Frames and doesn't know how many frames the loaded animation has)